src/sonic-host-services-data/debian/sonic-host-services-data.hostcfgd.service

```diff
@@ -1,6 +1,7 @@
 [Unit]
 Description=Host config enforcer daemon
 Requires=updategraph.service
+After=interfaces-config.service
```

Comment:
>interfaces [](http://example.com/codeflow?start=6&length=10)

Since original there is a Redis access error, then wait after database service?

Reply:
It is already implicitly a dependency of database.service due to a requirement on updategraph.service. The issue is not due to hostcfgd starts when database is not up, this issue happens when in the middle the interfaces-config restarts networking service which makes connecting though ConfigDBConnector() fail since by default it tries to connec to redis through localhost:6379. We have to either change it to use unix socket or make it start after interfaces-config. I decided to do 2nd fix due to otherwise ConfigDBConnector could use unix socket by default but if there was some motivation to make it use localhost by default then I could hit other problems.

Reply:
@qiluo-msft do you mean change the default to unix socket for ConfigDBConnector() or single hostcfgd?